rtgui/previewhandler.cc

```diff
@@ -204,27 +204,19 @@ Glib::RefPtr<Gdk::Pixbuf> PreviewHandler::getRoughImage (int x, int y, int w, in
             h = image->getHeight() * totalZoom;
         }
 
-        int ix = x * zoom;
-        int iy = y * zoom;
+        x *= zoom;
+        y *= zoom;
 
-        if (ix < 0) {
-            ix = 0;
+        if ((x + w) / totalZoom > previewImg->get_width()) {
```

Comment:
Ok, replaced by
```
w = rtengine::LIM<int>(w, 0, int(double(previewImg->get_width()) * totalZoom) - x);
```
I hope you don't mind the manual cast, it let me sleep at night better ;)